src/app/static/src/app/components/genericChart/GenericChart.vue

```diff
@@ -211,17 +211,21 @@
                 }
             },
             async updateDataSource(dataSourceId: string, datasetId: string) {
-                this.dataSourceSelections[dataSourceId]!.datasetId = datasetId;
+                //clear current datasource filter selections while we fetch & show spinner
+                this.updateSelectedFilterOptions(dataSourceId, null);
                 await this.ensureDataset(datasetId);
                 this.setDataSourceDefaultFilterSelections(dataSourceId, datasetId);
             },
             setDataSourceDefaultFilterSelections(dataSourceId: string, datasetId: string) {
                 const selectedFilterOptions = this.datasets[datasetId].metadata.defaults.selected_filter_options;
                 if (selectedFilterOptions) {
-                    this.updateSelectedFilterOptions(dataSourceId, {...selectedFilterOptions});
+                    this.dataSourceSelections[dataSourceId] = {
+                        datasetId,
+                        selectedFilterOptions
+                    };
```

Comment:
Update the selections object atomically here rather than separately (which caused potential propagation issues) - assume that the dataset id should be updated as well as the filter selections